robocorp-code/vscode-client/src/activities.ts

```diff
@@ -156,19 +156,28 @@ export async function resolveInterpreter(targetRobot: string): Promise<ActionRes
             "robot.resolveInterpreter",
             targetRobot
         );
+        if (interpreter === null || (typeof interpreter === "string" && interpreter === "null")) {
+            throw Error("Interpreter not found. Retrying call...");
+        }
+        vscode.window.showInformationMessage(`Environment built & cached. Python interpreter loaded.`);
         return { "success": true, "message": "", "result": interpreter };
     } catch (error) {
         // We couldn't resolve with the robotframework language server command, fallback to the robocorp code command.
         try {
-            let result: ActionResult<InterpreterInfo | undefined> = await commands.executeCommand(
+            let interpreter: ActionResult<InterpreterInfo | undefined> = await commands.executeCommand(
                 roboCommands.ROBOCORP_RESOLVE_INTERPRETER,
                 {
                     "target_robot": targetRobot,
                 }
             );
-            return result;
+            if (interpreter === null || (typeof interpreter === "string" && interpreter === "null")) {
+                throw Error("Interpreter not found");
+            }
+            vscode.window.showInformationMessage(`Environment built & cached. Python interpreter loaded.`);
```

Comment:
A message should not be shown if everything is correct here. You may want to move it to `ROBOCORP_PACKAGE_ENVIRONMENT_REBUILD` though.